// @supabase/auth-ui-react/dist/esm/src/components/Auth/interfaces/MagicLink.js
import { __awaiter } from '../../../../_virtual/_tslib.js';
import { jsx, jsxs } from 'react/jsx-runtime';
import { useState } from 'react';
import { VIEWS } from '../../../constants.js';
import { Anchor } from '../../UI/Anchor.js';
import { Button } from '../../UI/Button.js';
import { Container } from '../../UI/Container.js';
import '../../UI/Divider.js';
import { Input } from '../../UI/Input.js';
import { Label } from '../../UI/Label.js';
import { Message } from '../../UI/Message.js';

function MagicLink(_ref) {
  let {
    setAuthView,
    supabaseClient,
    redirectTo,
    i18n,
    appearance,
    showLinks
  } = _ref;

  var _a, _b, _c, _d;

  const [email, setEmail] = useState('');
  const [error, setError] = useState('');
  const [message, setMessage] = useState('');
  const [loading, setLoading] = useState(false);

  const handleMagicLinkSignIn = e => __awaiter(this, void 0, void 0, function* () {
    e.preventDefault();
    setError('');
    setMessage('');
    setLoading(true);
    const {
      error
    } = yield supabaseClient.auth.signInWithOtp({
      email,
      options: {
        emailRedirectTo: redirectTo
      }
    });
    if (error) setError(error.message);else setMessage('Check your email for the magic link');
    setLoading(false);
  });

  return jsx("form", Object.assign({
    id: "auth-magic-link",
    onSubmit: handleMagicLinkSignIn
  }, {
    children: jsxs(Container, Object.assign({
      gap: "large",
      direction: "vertical",
      appearance: appearance
    }, {
      children: [jsxs(Container, Object.assign({
        gap: "large",
        direction: "vertical",
        appearance: appearance
      }, {
        children: [jsxs("div", {
          children: [jsx(Label, Object.assign({
            appearance: appearance
          }, {
            children: (_a = i18n === null || i18n === void 0 ? void 0 : i18n.magic_link) === null || _a === void 0 ? void 0 : _a.email_input_label
          })), jsx(Input, {
            type: "email",
            placeholder: (_b = i18n === null || i18n === void 0 ? void 0 : i18n.magic_link) === null || _b === void 0 ? void 0 : _b.email_input_placeholder,
            onChange: e => setEmail(e.target.value),
            appearance: appearance
          })]
        }), jsx(Button, Object.assign({
          color: "primary",
          type: "submit",
          loading: loading,
          appearance: appearance
        }, {
          children: (_c = i18n === null || i18n === void 0 ? void 0 : i18n.magic_link) === null || _c === void 0 ? void 0 : _c.button_label
        }))]
      })), showLinks && jsx(Anchor, Object.assign({
        href: "#auth-sign-in",
        onClick: e => {
          e.preventDefault();
          setAuthView(VIEWS.SIGN_IN);
        },
        appearance: appearance
      }, {
        children: (_d = i18n === null || i18n === void 0 ? void 0 : i18n.sign_in) === null || _d === void 0 ? void 0 : _d.link_text
      })), message && jsx(Message, Object.assign({
        appearance: appearance
      }, {
        children: message
      })), error && jsx(Message, Object.assign({
        color: "danger",
        appearance: appearance
      }, {
        children: error
      }))]
    }))
  }));
}

export { MagicLink };
