// @supabase/auth-ui-react/dist/cjs/common/lib/Localization/de_informal.json.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var sign_up = {
	email_label: "E-Mail Adresse",
	password_label: "Passwort erstellen",
	email_input_placeholder: "Deine E-Mail Adresse",
	password_input_placeholder: "Dein Passwort",
	button_label: "Registrieren",
	social_provider_text: "Anmelden mit",
	link_text: "Hast du noch kein Konto? Registrieren"
};
var sign_in = {
	email_label: "E-Mail Adresse",
	password_label: "Passwort erstellen",
	email_input_placeholder: "Deine E-Mail Adresse",
	password_input_placeholder: "Dein Passwort",
	button_label: "Anmelden",
	social_provider_text: "Anmelden mit",
	link_text: "Hast du bereits ein Konto? Anmelden"
};
var magic_link = {
	email_input_label: "E-Mail Adresse",
	email_input_placeholder: "Deine E-Mail Adresse",
	button_label: "Magischen Link senden",
	link_text: "Einen magischen Link per E-Mail versenden"
};
var forgotten_password = {
	email_label: "E-Mail Adresse",
	password_label: "Dein Passwort",
	email_input_placeholder: "Deine E-Mail Adresse",
	button_label: "Anweisungen zum Zurücksetzen des Passworts senden",
	link_text: "Passwort vergessen?"
};
var update_password = {
	password_label: "Neues Passwort",
	password_input_placeholder: "Dein neues Passwort",
	button_label: "Passwort aktualisieren"
};
var de_informal = {
	sign_up: sign_up,
	sign_in: sign_in,
	magic_link: magic_link,
	forgotten_password: forgotten_password,
	update_password: update_password
};

exports["default"] = de_informal;
exports.forgotten_password = forgotten_password;
exports.magic_link = magic_link;
exports.sign_in = sign_in;
exports.sign_up = sign_up;
exports.update_password = update_password;
