// @supabase/auth-ui-react/dist/cjs/common/lib/Localization/ja.json.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var sign_up = {
	email_label: "電子メールアドレス",
	password_label: "パスワードを作成",
	email_input_placeholder: "Your email address",
	password_input_placeholder: "Your password",
	button_label: "サインアップ",
	social_provider_text: "に登録する",
	link_text: "アカウントをお持ちではありませんか？サインアップ"
};
var sign_in = {
	email_label: "電子メールアドレス",
	password_label: "あなたのパスワード",
	email_input_placeholder: "Your email address",
	password_input_placeholder: "Your password",
	button_label: "サインイン",
	social_provider_text: "に登録する",
	link_text: "Already have an account? Sign in"
};
var magic_link = {
	email_input_label: "Email address",
	email_input_placeholder: "Your email address",
	button_label: "Send Magic Link",
	link_text: "Send a magic link email"
};
var forgotten_password = {
	email_label: "Email address",
	password_label: "Your Password",
	email_input_placeholder: "Your email address",
	button_label: "Send reset password instructions",
	link_text: "パスワードをお忘れの方"
};
var update_password = {
	password_label: "New password",
	password_input_placeholder: "Your new password",
	button_label: "Update password"
};
var ja = {
	sign_up: sign_up,
	sign_in: sign_in,
	magic_link: magic_link,
	forgotten_password: forgotten_password,
	update_password: update_password
};

exports["default"] = ja;
exports.forgotten_password = forgotten_password;
exports.magic_link = magic_link;
exports.sign_in = sign_in;
exports.sign_up = sign_up;
exports.update_password = update_password;
